libdebug/data/memory_map.py

```diff
@@ -64,6 +68,42 @@ def base(self: MemoryMap) -> int:
         """Alias for the start address of the memory map."""
         return self.start
 
+    @property
+    def content(self: MemoryMap) -> bytes:
+        """The content of the memory map, used for snapshotted pages."""
+        if self._content is None:
+            raise ValueError("The content field is only available for snapshot pages.")
+
+        return self._content
+
+    def is_same_identity(self: MemoryMap, other: MemoryMap) -> bool:
+        """Check if the memory map corresponds to another memory map."""
+        return self.start == other.start and self.backing_file == other.backing_file
+
     def __repr__(self: MemoryMap) -> str:
         """Return the string representation of the memory map."""
-        return f"MemoryMap(start={hex(self.start)}, end={hex(self.end)}, permissions={self.permissions}, size={hex(self.size)}, offset={hex(self.offset)}, backing_file={self.backing_file})"
+        str_repr =  f"MemoryMap(start={hex(self.start)}, end={hex(self.end)}, permissions={self.permissions}, size={hex(self.size)}, offset={hex(self.offset)}, backing_file={self.backing_file})"
+
+        if self._content is not None:
+            str_repr = str_repr[:-1] + ", content=...)"
+
+        return str_repr
+
+    def __eq__(self, value: object) -> bool:
+        """Check if this MemoryMap is equal to another object."""
+        if not isinstance(value, MemoryMap):
+            return False
+
+        # Check if the content is available and if it is the same
+        should_compare_content = self._content is not None and value._content is not None
+        same_content = not should_compare_content or self._content == value._content
```

Comment:
Yes, when comparing base snapshots and full snapshots, you can't say the page is different because you don't have the content. So we can either compare the rest of the field (like it is now) or raise an exception to disallow comparison of memory pages that do not have the same content type (one is NoneType, the other is bytes).